core/lib/mini_merkle_tree/src/tests.rs

```diff
@@ -156,24 +156,81 @@ fn verify_merkle_proof(
     assert_eq!(hash, merkle_root);
 }
 
+fn verify_interval_merkle_proof(
+    items: &[[u8; 88]],
+    mut head_index: usize,
+    left_path: &[H256],
+    right_path: &[H256],
+    merkle_root: H256,
+) {
+    assert_eq!(left_path.len(), right_path.len());
+
+    let hasher = KeccakHasher;
+    let mut hashes: Vec<_> = items.iter().map(|item| hasher.hash_bytes(item)).collect();
+    let mut level_len = hashes.len();
+
+    for (left_item, right_item) in left_path.iter().zip(right_path.iter()) {
```

Comment:
This can be simplified using the same approach I've mentioned above, especially since performance is not a concern for tests: Construct hashes on a certain level as `left_path[level]? || hashes || right_path[level]?`, so that elements start from an even index and contain all data to create the next level by zipping hashes pairwise.